lib/ansible/plugins/lookup/mongodb.py

```diff
@@ -163,49 +164,20 @@ def convert_mongo_result_to_valid_json(self, result):
             return u"{0}".format(result)
 
     def run(self, terms, variables, **kwargs):
-
+        try:
+            return self._run_helper(terms)
+        except Exception as e:
+            print(u"There was an exception on the mongodb_lookup: {0}".format(to_native(e)))
+            raise e
+
+    def _run_helper(self, terms):
+        if not pymongo_found:
+            raise AnsibleError(u"pymongo is required in the master node (this machine) for mongodb lookup.")
```

Comment:
I would prefer not to, because this is a lookup module and therefore it must be in master node instead of the target node. Your message may confuse users who are not experts in ansible.

Reply:
in this case, you should use `primary` instead of `master`

Reply:
understood.! fixed!

Reply:
`control node` = Where ansible-playbook is run
`remote node` = system to configure